timelock-impl/src/main/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceFactory.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import java.util.Optional;
+
+public interface AsyncTimelockServiceFactory {
+    AsyncTimelockService get(String namespace, Optional<String> userAgent);
```

Comment:
this is really `getOrCreate` but you need to pierce through a couple of layers to see that, so found no need to name accordingly here